### PR TITLE
Update and refactor policy and policy_groups endpoints

### PR DESCRIPTION
**Pull request by jkeiser**

### CODE REVIEW

**Comment by danielsdeleo** [comment]
LGTM :+1:

File: bin/chef-zero [hunk 1]
```diff
@@ -55,6 +55,11 @@ OptionParser.new do |opts|
     options[:log_file] = value
   end
 
+  opts.on("--enterprise", "Whether to run in enterprise mode") do |value|
```
Comment (tyler-ball) [inline bin/chef-zero]: I realize this PR is just the chef-zero part, but will we need to push this config option up to `chef-client -z`?

File: lib/chef_zero/endpoints/policy_group_policy_endpoint.rb [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+    # in the data store, this REST path actually stores the revision ID of ${policy_name} that's currently
+    # associated with ${policy_group}.
+    class PolicyGroupPolicyEndpoint < RestBase
+
+      # GET /organizations/ORG/policy_groups/GROUP/policies/NAME
+      def get(request)
+        policy_name = request.rest_path[5]
+
+        # fetch /organizations/{organization}/policies/{policy_name}/revisions/{revision_id}
+        revision_id = parse_json(get_data(request))
```
Comment (tyler-ball) [inline lib/chef_zero/endpoints/policy_group_policy_endpoint.rb]: Does this parse the GET body?  If so, I'm surprised we send a GET request with data in the body

File: lib/chef_zero/rspec.rb [hunk 3]
```diff
@@ -149,6 +149,14 @@ def org_member(*usernames)
         before(chef_server_options[:server_scope]) { org_member(*usernames) }
       end
 
+      def policy(name, data, &block)
+        before(chef_server_options[:server_scope]) { node(name, data, &block) }
```
Comment (tyler-ball) [inline lib/chef_zero/rspec.rb]: Is this supposed to be calling `node` here or `policy`?

**Comment by tyler-ball** [comment]
Can we uncomment the [pedant policies](https://github.com/chef/chef-zero/blob/0344829524c43f2872fe54f8b89198d04441b44e/spec/run_oc_pedant.rb#L86) test yet?

Some questions, but overall :+1:

File: lib/chef_zero/endpoints/policy_group_policy_endpoint.rb (hunk 2 above)
Comment (randomcamel) [inline lib/chef_zero/endpoints/policy_group_policy_endpoint.rb]: The latter.

File: bin/chef-zero (hunk 1 above)
Comment (jkeiser) [inline bin/chef-zero]: It's really a separate question. chef-client -z doesn't support full-on enterprise mode yet (there are separate PRs for that).

File: lib/chef_zero/rspec.rb (hunk 3 above)
Comment (jkeiser) [inline lib/chef_zero/rspec.rb]: policy, good catch!

**Comment by randomcamel** [comment]
@tyler-ball Not yet: those are the tests to skip for ChefFS, and that's PR https://github.com/chef/chef/pull/4254, which depends on stuff in _this_ PR, so this has to go first.

**Comment by tyler-ball** [comment]
@randomcamel :+1:

**Pull request merged**